app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardClustersStep/MappingWizardClustersStep.js

```diff
@@ -26,6 +39,103 @@ class MappingWizardClustersStep extends React.Component {
     fetchTargetClustersAction(fetchTargetClustersUrl);
   }
 
+  selectSourceCluster(sourceCluster) {
+    this.setState(prevState => {
+      const isAlreadySelected = prevState.selectedSourceClusters.some(
+        cluster => cluster.id === sourceCluster.id
+      );
+      if (isAlreadySelected) {
+        return {
+          selectedSourceClusters: prevState.selectedSourceClusters.filter(
+            cluster => cluster.id !== sourceCluster.id
+          )
+        };
+      }
+      return {
+        selectedSourceClusters: [
+          ...prevState.selectedSourceClusters,
+          sourceCluster
+        ]
+      };
+    });
+  }
+
+  selectTargetCluster(targetCluster) {
+    this.setState(() => ({ selectedTargetCluster: targetCluster }));
+  }
+
+  addMapping() {
```

Comment:
we may need to bubble up redux  `addMapping(), removeMapping(), removeAll()` to `MappingWizardReducers.js` (doesn't exist yet) so that those mappings are available on Step 3 for Datastores (which I'll have a PR up for soon)...and then for the final POST...

I'm also on the fence about ModalWizard.StateProvider .. kind of think it would be cleaner to just have all MappingWizard state reduced in one place and connected to Redux. Will hash this out tomorrow w/ you..